File: Cofeeshop.py
class CoffeeShop:
    def __init__(self, name):
        self.name = name
        self.menu = {}
        self.orders = []

    def add_item_to_menu(self, item, price):
        self.menu[item] = price

    def show_menu(self):
        print(f"Menu at {self.name}:")
        for item, price in self.menu.items():
            print(f"{item}: ${price:.2f}")

    def take_order(self, item):
        if item in self.menu:
            self.orders.append(item)
            print(f"Order added: {item}")
        else:
            print(f"Sorry, {item} is not on the menu.")

    def show_orders(self):
        print("Current Orders:")
        for order in self.orders:
            print(order)

    def calculate_total(self):
        total = sum(self.menu[item] for item in self.orders)
        print(f"Total: ${total:.2f}")
        return total

    def clear_orders(self):
        self.orders = []
        print("All orders cleared.")

# Example usage
if __name__ == "__main__":
    shop = CoffeeShop("Java Beans")
    shop.add_item_to_menu("Espresso", 2.50)
    shop.add_item_to_menu("Latte", 3.50)
    shop.add_item_to_menu("Cappuccino", 3.00)

    shop.show_menu()
    shop.take_order("Espresso")
    shop.take_order("Latte")
    shop.show_orders()
    shop.calculate_total()
    shop.clear_orders()

File: CofeeShop.js
// Coffee Shop Program

class CoffeeShop {
    constructor(name) {
        this.name = name;
        this.menu = [];
        this.orders = [];
    }

    addMenuItem(item) {
        this.menu.push(item);
    }

    placeOrder(itemName) {
        const item = this.menu.find(menuItem => menuItem.name === itemName);
        if (item) {
            this.orders.push(item);
            return `${itemName} has been added to your order.`;
        } else {
            return `Sorry, ${itemName} is not available on the menu.`;
        }
    }

    fulfillOrder() {
        if (this.orders.length > 0) {
            const fulfilledOrder = this.orders.shift();
            return `Order for ${fulfilledOrder.name} is ready!`;
        } else {
            return "No orders to fulfill.";
        }
    }

    listOrders() {
        return this.orders.map(order => order.name);
    }

    dueAmount() {
        return this.orders.reduce((total, order) => total + order.price, 0).toFixed(2);
    }

    cheapestItem() {
        if (this.menu.length === 0) return "Menu is empty.";
        return this.menu.reduce((cheapest, item) => item.price < cheapest.price ? item : cheapest).name;
    }

    drinksOnly() {
        return this.menu.filter(item => item.type === "drink").map(item => item.name);
    }

    foodOnly() {
        return this.menu.filter(item => item.type === "food").map(item => item.name);
    }
}

// Example usage
const myCoffeeShop = new CoffeeShop("Java Haven");

myCoffeeShop.addMenuItem({ name: "Espresso", type: "drink", price: 3.0 });
myCoffeeShop.addMenuItem({ name: "Latte", type: "drink", price: 4.5 });
myCoffeeShop.addMenuItem({ name: "Croissant", type: "food", price: 2.5 });

console.log(myCoffeeShop.placeOrder("Latte"));
console.log(myCoffeeShop.placeOrder("Croissant"));
console.log(myCoffeeShop.fulfillOrder());
console.log("Current Orders:", myCoffeeShop.listOrders());
console.log("Total Due:", myCoffeeShop.dueAmount());
console.log("Cheapest Item:", myCoffeeShop.cheapestItem());
console.log("Drinks Only:", myCoffeeShop.drinksOnly());
console.log("Food Only:", myCoffeeShop.foodOnly());